### ISSUE TITLE
Error while executing the amplify push/pull command using pre-existing lambdas and layers in a multi-environment configuration.

### ISSUE THREAD

**Issue by fkoner** [aws-amplify/amplify-cli#5250]
**Describe the bug**
Error while executing the amplify push/pull command using pre-existing lambdas and layers in a multi-environment configuration. The following error occurs.

```
2020-09-03T22:58:22.587Z [WARNING]: ✖ An error occurred when pushing the resources to the cloud
2020-09-03T22:58:34.517Z [WARNING]: ✖ There was an error initializing your environment.
2020-09-03T22:58:35.269Z [INFO]: Failed to pull the backend.
2020-09-03T22:58:35.271Z [INFO]: File at path: '/codebuild/output/src827925238/src/ppm-trademark-console/amplify/backend/function/openPayModules/layer-runtimes.json' does not exist
2020-09-03T22:58:35.291Z [INFO]: Error: File at path: '/codebuild/output/src827925238/src/ppm-trademark-console/amplify/backend/function/openPayModules/layer-runtimes.json' does not exist
                                 at Function.JSONUtilities.readJson (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/src/jsonUtilities.ts:25:15)
                                 at Object.exports.getLayerRuntimes (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-function/src/provider-utils/awscloudformation/utils/layerRuntimes.ts:12:24)
                                 at new LayerState (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-function/src/provider-utils/awscloudformation/utils/layerParams.ts:89:21)
                                 at layerName (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-function/src/provider-utils/awscloudformation/utils/layerParams.ts:247:12)
                                 at ensureLayerVersion (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-function/src/provider-utils/awscloudformation/utils/packageLayer.ts:81:54)
                                 at packageLayer (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-function/src/provider-utils/awscloudformation/utils/packageLayer.ts:18:9)
                                 at packageResource (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/lib/push-resources.js:230:68)
                                 at packageResources (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/lib/push-resources.js:302:19)
                                 at Object.run (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/lib/push-resources.js:46:11)
                                 at Object.pushResources (/root/.nvm/versions/node/v10.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-provider-awscloudformation/index.js:40:25)
```

**Amplify CLI Version**
4.29.0

**To Reproduce**
Have an application with a function (lambda) and this function with a layer configured. Its required to have created this application, lambda and layer with a version prior to 4.29.0 (for example 4.27.0), upgrading to version 4.29.0 and trying to run an amplify **_push/pull_** command in the console with the **_amplifyPush --simple_**  and the error appears. In our application we have 3 functions that share one layer, and two other functions that share another completely different layer.

**Expected behavior**
Amplify pull command should retrieve changes for pre-existing lambdas and layers in a multi-environment setup

**Screenshots**
![aws-cli-multi-env-setup-lambda-layers-issue](https://user-images.githubusercontent.com/2480544/92205421-453ffa00-ee4b-11ea-963a-b90932cab330.png)

**Desktop (please complete the following information):**
 - OS: [e.g. Mac/Windows/Ubuntu]
 - Node Version. 10.20.1

**Additional context**
 I don't know if it is related to the amplify console, or to changes made in the command line.
The following commit is the cause:
https://github.com/aws-amplify/amplify-cli/commit/4af13634bcdd58511712249e6774fc9f287c9ef5

Specifically this file:

https://github.com/aws-amplify/amplify-cli/blob/4af13634bcdd58511712249e6774fc9f287c9ef5/packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerRuntimes.ts#L8

**Thanks in advance**
**_Paco_**

**Comment by fkoner** [comment]
@jhockett @kennandavison

Is it required to create this file?

layer-runtimes.json

With the folowing content?

```json
[
  {
    "value": "nodejs",
    "name": "NodeJS",
    "layerExecutablePath": "nodejs/node_modules",
    "cloudTemplateValue": "nodejs10.x"
  }
]
```

**Comment by fkoner** [comment]
@jhockett 

I found the following comment from you:

https://github.com/aws-amplify/amplify-cli/issues/4886#issuecomment-686798972

¿Do I need to upgrade my lambda layers with these new amplify release (4.29.0)?

**Comment by jhockett** [comment]
Hi @fkoner, thanks for reporting this. This looks like a migration bug.

> Is it required to create this file?
> 
> layer-runtimes.json
> 
> With the folowing content?
> 
> ```json
> [
>   {
>     "value": "nodejs",
>     "name": "NodeJS",
>     "layerExecutablePath": "nodejs/node_modules",
>     "cloudTemplateValue": "nodejs10.x"
>   }
> ]
> ```

Creating the `layer-runtimes.json` file as you described should be the easiest way to get unblocked until this is patched.

> ¿Do I need to upgrade my lambda layers with these new amplify release (4.29.0)?

You don't have to, but I recommend it because layers created before 4.29.0 can have issues if you decide to use multiple environments. If you create a new Lambda layer to replace the existing one, that will also unblock you.

**Comment by fkoner** [comment]
Hi @jhockett.

Thanks for your support. As you recommend, those are temporary solutions, am I correct?

The ideal solution is for the amplify push / pull commands to realize that this file (**_layer-runtimes.json_**) is not present and create it for you, right? Because migrating dozens of layers is not ideal.

**Regards**

**Comment by fkoner** [comment]
Hi again @jhockett 

I have created a new layer, and I noticed that my _backend-config.json_ file updates the pre-existing layers, as follows:

```
{
  "function": {
    "openPayModules": {
      "runtimes": [
        {
          "value": "nodejs",
          "name": "NodeJS",
          "layerExecutablePath": "nodejs/node_modules",
          "cloudTemplateValue": "nodejs12.x"
        }
      ],
      "layerVersionMap": {
        "1": {
          "permissions": [
            {
              "type": "private"
            }
          ]
        }
      },
      "providerPlugin": "awscloudformation",
      "service": "LambdaLayer",
      "build": true
    }
  }
}
```
I see that by default it associates version 1 to the pre-existing layers. Should this work like this?

Or should you ask us what version of the layer you want to assign?

Another thing that I have noticed, and I do not know if it has happened to you, is that every time you run an amplify push, the version of the layer increases, and there comes a point where the layer stops working I do not know if it is due to so many versions. In fact, I don't know if this should be the correct behavior, that is, that each push updates the version number, even when there is no change.

On the issue that layers stop working I can't find an answer, they just stop working at some point, especially when I have code stored in the /opt directory. I think I'll open an issue with this point.

**Comment by jeremie-olivier** [comment]
> @jhockett @kennandavison
> 
> Is it required to create this file?
> 
> layer-runtimes.json
> 
> With the folowing content?
> 
> ```json
> [
>   {
>     "value": "nodejs",
>     "name": "NodeJS",
>     "layerExecutablePath": "nodejs/node_modules",
>     "cloudTemplateValue": "nodejs10.x"
>   }
> ]
> ```
> 
> **Thanks in advance**
> **_Paco_**

This worked for me. Thanks

**Comment by jhockett** [comment]
A fix for this has been released in v4.29.1
